raiden/transfer/secret_registry.py

```diff
@@ -14,16 +20,28 @@ def events_for_onchain_secretreveal(
     secret: Secret,
     expiration: BlockExpiration,
     block_hash: BlockHash,
+    target_state: TargetTransferState = None,
+    pseudo_random_generator: random.Random = None
 ) -> List[Event]:
-    events: List[Event] = list()
-
     if not isinstance(secret, T_Secret):
         raise ValueError("secret must be a Secret instance")
 
     if get_status(channel_state) in CHANNEL_STATES_PRIOR_TO_CLOSED + (CHANNEL_STATE_CLOSED,):
-        reveal_event = ContractSendSecretReveal(
-            expiration=expiration, secret=secret, triggered_by_block_hash=block_hash
-        )
-        events.append(reveal_event)
+        if not channel_state.is_light_channel:
+            return [
+                ContractSendSecretReveal(
+                    expiration=expiration,
+                    secret=secret,
+                    triggered_by_block_hash=block_hash,
+                )
+            ]
+        if target_state is not None:
```

Comment:
This could be replaced by ``if target_state``

Reply:
In Python, I've been always confused of the prefered way. [Here](https://stackoverflow.com/questions/7152441/python-if-not-val-vs-if-val-is-none) they say that using 'is' is more pythonic, but don't understand clearly the reasons.

Maybe sir @mortelli could illustrate us a bit more

Reply:
Well basically the if statement checks for empty values like None, "", empty list etc. So the statement if something is not None is equivalent to if something.

Reply:
I know, it maps them to booleans. But thats not what I'm discussing

Reply:
my personal preference would be in agreement to what @JONAF2103 is saying.

if it was anything else other than `None`, i would leave it the way it is